test/new-e2e/utils/e2e/e2e.go

```diff
@@ -112,43 +100,42 @@ func (suite *Suite[Env]) SetupSuite() {
 	err := client.CheckEnvStructValid[Env]()
 	require.NoError(err)
 
-	credentialsManager := credentials.NewManager()
-	apiKey, err := credentialsManager.GetCredential(credentials.AWSSSMStore, "agent.ci.dev.apikey")
-	require.NoError(err)
-
-	sshKey, err := credentialsManager.GetCredential(credentials.AWSSSMStore, "agent.ci.awssandbox.ssh")
+	env, _, upResult, err := createEnv(suite, suite.stackDef)
 	require.NoError(err)
 
-	suite.auth.SSHKey = sshKey
-	suite.add(config.DDAgentConfigNamespace, config.DDAgentAPIKeyParamName, apiKey)
-	suite.add(config.DDInfraConfigNamespace, aws.DDInfraDefaultKeyPairParamName, "agent-ci-sandbox")
-	suite.add(config.DDInfraConfigNamespace, aws.DDInfraDefaultInstanceTypeParamName, "t3.large")
-	suite.add(config.DDInfraConfigNamespace, aws.DDInfraDefaultARMInstanceTypeParamName, "m6g.medium")
-	env, stack, upResult, err := createEnv(suite, suite.stackDef)
-	require.NoError(err)
-	suite.fullStackName = stack.Name()
 	suite.Env = env
 	err = client.CallStackInitializers(&suite.auth, env, upResult)
 	require.NoError(err)
 }
 
+// HandleStats method is run after all the tests in the suite have been run.
+// and after TearDownSuite has been run.
+// This function is called by [testify Suite].
+//
+// [testify Suite]: https://pkg.go.dev/github.com/stretchr/testify/suite
+func (suite *Suite[Env]) HandleStats(string, stats *suite.SuiteInformation) {
```

Comment:
What is the reason to not use `TearDownSuite` which is more common.

Reply:
Because you cannot provide the same feature. In `TearDownSuite` the pass/fail info is not available.